priv/posts/nodejs-cve-to-eol-reasoning.md

```diff
@@ -0,0 +1,72 @@
+
+- **CVE-2025-23087** targets Node.js v17 and all prior versions (starting with v0.x).
+- **CVE-2025-23088** targets Node.js v19.
+- **CVE-2025-23089** targets Node.js v21.
+
+Moving forward, Node.js will continue issuing CVEs for subsequent EOL versions.
+
+This approach was previously discussed and announced on the [Node.js blog](https://nodejs.org/en/blog/vulnerability/upcoming-cve-for-eol-versions).
+The intent is to raise awareness among users that running an EOL version of Node.js is dangerous,
+as it **contains** numerous unpatched vulnerabilities.
+
+Unsurprisingly, the security community has debated whether issuing CVEs for EOL versions is necessary or
+philosophically acceptable. This post aims to clarify the Node.js perspective and the reasoning behind this decision.
+
+## Why Node.js Doesn't Check Reports Against EOL Versions
+
+Some critics argue that Node.js should evaluate security reports against EOL versions and include them in
+CVE version ranges. However, the answer to this is straightforward: Node.js is maintained by volunteers.
+Currently, there are over 20 EOL versions of Node.js, each with different dependencies, versions of V8, build methods,
+and supported platforms. Checking whether every vulnerability affects these versions is neither sustainable nor practical.
+
+Setting a threshold for the last N LTS EOL versions could also lead to errors. When a vulnerability report is submitted to
+Node.js via HackerOne, the team ensures the issue is applicable across all active release lines but
+excludes EOL versions entirely. Attempting to include EOL versions could result in false negatives, as vulnerabilities may
+manifest differently depending on the version and its dependencies (e.g., OpenSSL).
+
+In other words, evaluating reports against EOL versions would require resources far beyond the project’s capacity.
+Node.js operates primarily through volunteer efforts, and even with its existing funding,
+resources are focused on keeping current releases secure.
+
+As Matteo Collina summarized:
+
+> Why did the Node.js project emit a CVE for all EOL releases? Because we don’t have the resources to evaluate every single past release to know what is vulnerable.
+>
+> For all of you that are asking, Node.js is run by volunteers. We have enough funding to maintain the current releases safe, but nothing more.
+>
+> In other words, we are 100% certain that all those past releases of Node.js are vulnerable, or will soon be vulnerable. This CVE now will raise the awareness of this security risk to your organization.
+
+([Source: Matteo Collina on X](https://x.com/matteocollina/status/1882892694722101326))
+
+## The Node.js Intention with These CVEs
+
+Some security experts question whether CVEs are the "correct" method for informing users about EOL software.
+While we welcome philosophical debates, issuing CVEs was the most effective way to achieve our primary goal: reaching Node.js users directly.
```

Comment:
```suggestion
```

Reply:
Why @mhdawson ?

Reply:
I'd prefer we stick to saying what we are doing and why, not saying people are arguing with us and trying to argue back.